index.js

```diff
@@ -492,6 +498,31 @@ function getStyleFunction(glStyle, source, resolutions, onChange) {
           }
         }
 
+        if (type == 'Point' && 'circle-radius' in paint) {
+          ++stylesLength;
+          var cache_key = paint['circle-radius'] + '.' +
```

Comment:
`circle-radius`, `circle-stroke-color` and `circle-color` are functions, so you have to use e.g. `paint['circle-radius'](zoom)` for the cache key as well.